syft/frameworks/torch/mpc/przs.py

```diff
@@ -0,0 +1,154 @@
+from collections import defaultdict
+from syft.frameworks.torch.mpc.primitives import PrimitiveStorage
+from syft.generic.utils import remote, allow_command
+
+import torch
+import syft
+
+RING_SIZE = 2 ** 64
+ERR_MSG = "You must call PRZS.setup because the seeds where not shared between workers"
+
+
+class PRZS:
+    def __init__(self):
+        self.generators = {}
+
+    @property
+    def generators(self):
+        return self.__generators
+
+    @generators.setter
+    def generators(self, generators):
+        self.__generators = generators
+
+    @staticmethod
+    def setup(workers):
+        seed_max = 2 ** 32
+        paired_workers = list(zip(workers, workers[1:]))
+        paired_workers.append((workers[-1], workers[0]))
+
+        workers_ptr = defaultdict(dict)
+
+        for cur_worker, next_worker in paired_workers:
+            if cur_worker == syft.local_worker:
+                ptr = cur_worker.torch.randint(-seed_max, seed_max - 1, size=(1,))
+                ptr_next = ptr.send(next_worker)
+            else:
+                ptr = cur_worker.remote.torch.randint(-seed_max, seed_max - 1, size=(1,))
+                ptr_next = ptr.copy().move(next_worker)
+
+            workers_ptr[cur_worker]["cur_seed"] = ptr
+            workers_ptr[next_worker]["prev_seed"] = ptr_next
+
+        for worker, seeds in workers_ptr.items():
+            cur_seed = seeds["cur_seed"]
+            prev_seed = seeds["prev_seed"]
+            func_remote = remote(_initialize_generators, location=worker)
+            func_remote(cur_seed, prev_seed)
+
+
+def get_random(name_generator, shape, worker):
```

Comment:
Currently, this function is used only for getting a random number from a specific generator (used only for testing - we might remove it if we find no need for it)